src/main/java/com/nekomaster1000/infernalexp/entities/BlindsightEntity.java

```diff
@@ -111,7 +113,17 @@ protected void jump() {
         jumpDuration = 10;
         jumpTicks = 0;
 
-        super.jump();
+        float f = this.getJumpUpwardsMotion() + (rand.nextFloat() * 0.7F);
+
+        Vector3d vector3d = this.getMotion();
+        this.setMotion(vector3d.x, f, vector3d.z);
+        if (this.isSprinting()) {
+            float f1 = this.rotationYaw * ((float)Math.PI / 180F);
+            this.setMotion(this.getMotion().add((-MathHelper.sin(f1) * 0.2F), 0.0D, (MathHelper.cos(f1) * 0.2F)));
+        }
+
+        this.isAirBorne = true;
+        net.minecraftforge.common.ForgeHooks.onLivingJump(this);
```

Comment:
why

Reply:
Shut up swan you’re not a dev

Reply:
It's the code from super.jump(), but moved up to allow us to modify it easily. I added a random value to the calculation of f to get a range of jump heights for the Blindsight to give it a little bit more character.